service/server.py

```diff
@@ -9,16 +13,78 @@
 
 logger = logging.getLogger(__name__)
 register_title_api = app.config['REGISTER_TITLE_API']
+login_api = app.config['LOGIN_API']
+login_json = '{{"credentials":{{"user_id":"{}","password":"{}"}}}}'
+forward_slash = '/'
+unauthorised_wording = 'There was an error with your Username/Password combination. Please try again'
 google_analytics_api_key = app.config['GOOGLE_ANALYTICS_API_KEY']
 
-# TODO Create a proper secret key and store it securely
-app.secret_key = 'a_secret_key'
+LOGGER = logging.getLogger(__name__)
+
+
+class User():
+    def __init__(self, username):
+        self.userid = username
+
+    def get_id(self):
+        return self.userid
+
+    def is_authenticated(self):
+        return True
+
+    def is_active(self):
+        return True
+
+    def is_anonymous(self):
+        return False
+
+@login_manager.user_loader
+def load_user(userid):
+    return User(userid)
 
 @app.route('/', methods=['GET'])
 def home():
-    return render_template('home.html', asset_path = '../static/')
+    return render_template('home.html', asset_path='../static/')
+
+@app.route('/login', methods=['GET'])
+def signin_page():
+    # csrf_enabled = False for development environment only
+    return render_template('display_login.html', asset_path='../static/', form=SigninForm(csrf_enabled=False))
+
+@app.route('/login', methods=['POST'])
+def signin():
+    form = SigninForm(csrf_enabled=False)
+    # need to record the URL user was trying to hit before being redirected to login page
+    redirection_url = request.args.get('next') or 'title-search'
+    redirection_url = redirection_url.replace(forward_slash, '')
+    if not form.validate():
+        # entered details from login form incorrect so redirect back to same page with error messages
+        return render_template('display_login.html', asset_path='../static/', next=redirection_url, form=form)
+    else:
+        username = form.username.data
+        # form has correct details. Now need to check authorisation
+        authorised = get_login_auth(username, form.password.data)
+        if authorised:
+            login_user(User(username))
```

Comment:
We need to set REMEMBER_COOKIE_DURATION config value to something shorter than 365 days (default). Ideally this should be configurable.